assets/src/edit-story/elements/shared/useAverageColor.js

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import ColorThief from 'colorthief';
+
+/**
+ * WordPress dependencies
+ */
+import { useRef, useCallback, useLayoutEffect } from '@wordpress/element';
+
+function useAverageColor( ref, onAverageColor ) {
+	const thief = useRef( new ColorThief() );
+
+	const checkAverageColor = useCallback( ( ) => {
+		try {
+			onAverageColor( thief.current.getColor( ref.current ) );
+		} catch ( e ) {
+			// ColorThief fails for all-white images
+			//
+			// It's a "feature":
+			// https://github.com/lokesh/color-thief/pull/49
+			onAverageColor( [ 255, 255, 255 ] );
+		}
+	}, [ ref, onAverageColor ] );
```

Comment:
Typically the side-effect's handler is excluded from the effect's deps given that the expectation is to be able to inline the function. But I can be persuaded otherwise. Basically the requirement to use this hook now is to always use `useCallback` to construct the `onAverageColor`. Otherwise the `useLayoutEffect` will trigger each time.

Reply:
While I do agree with you to an extend, there also the principle of it. And we have `react-hooks/exhaustive-deps` to manage that principle - every external variable should be a dependency.

If I remove it, eslint reports:

```
eslint: error
react-hooks/exhaustive-deps - React Hook useCallback has a
missing dependency: 'onAverageColor'. Either include it or remove
the dependency array. If 'onAverageColor' changes too often,
find the parent component that defines it and wrap that definition
in useCallback.
```

So, I would have to either `useRef()` it or `eslint-disable` it. I'll go for the former for now.

But looking at it again, it does make sense to inline `checkAverageColor` in the `useEffect`.

Reply:
Done in b89eb6e. What do you think?

Reply:
Yeah. It seems good. Just FYI, I've been just passing optional `deps` from outside and disabling exhaustive-deps to keep it as close as possible to the `useEffect`. But let's try this way as well and see where we coalesce to.